src/core/runner.js

```diff
@@ -0,0 +1,377 @@
+import parser from './parser/parser';
+import results from './results';
+import DefaultVariableStorage from './default-variable-storage';
+import convertYarn from './convert-yarn-to-js';
+import types from './parser/nodes';
+
+const nodeTypes = types.types;
+
+class Runner {
+  constructor() {
+    this.noEscape = false;
```

Comment:
This is the one way in which the higher-level runner layer modifies the behavior of the core runner. Escaping is a destructive process and if we did it at the core level, we would not have the information to escape markup brackets. So we do that later on a line-by-line basis.